core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2015,6 +2015,26 @@ class WorkspaceService(
             .map(_.email)
         )
       )
+      inviteNotifications <- DBIO.from {
+        workspaceRequest.addUsers match {
+          case Some(acls) =>
+            collectMissingUsers(acls.map(_.email).toSet, ctx).flatMap { missingUsers =>
+              Future.sequence(missingUsers.map { email =>
+                samDAO.inviteUser(email, ctx).map { _ =>
+                  Notifications.WorkspaceInvitedNotification(
+                    WorkbenchEmail(email),
+                    WorkbenchUserId(ctx.userInfo.userSubjectId.value),
+                    NotificationWorkspaceName(workspaceRequest.namespace, workspaceRequest.name),
+                    workspaceId // TODO this should be bucketname but we don't have it yet?
```

Comment:
You could wait build the notifications until after the workspace is created. You still need to call `samDAO.inviteUser` before creating the workspace in sam though.

Reply:
I like Doug's suggestion which ensures you get all the information needed for the notification and ensures you are inviting the user to something that exists as well.

Reply:
Ok, moved sending invites to the end in https://github.com/broadinstitute/rawls/pull/3200/commits/a92a0cbdea3b0d936d320fa3603918e574f9cdd0